### DISCUSSION TITLE
Having one entity for each device per room assistant instance

### DISCUSSION THREAD

**Discussion by drfinn15** [mKeRix/room-assistant#744] [Q&A]
Hi there,
does anyone of you know whether there is a way to have one entity per each device and each room, so e.g. 2 devices + 2 RA instances = 4 entities in HA. My reason for this is that I would like to have the distance to each RA instance, not only to the closer one to do calculations with them on my own.
Thanks for every tip or idea!
BR, Finn

**Comment by benjycov** [comment]
Hi,

I am not sure, but perhaps if you didn't cluster the RA instances and had them reporting different IDs for the same device (for example Iphone_livingroom, iphone_kitchen), that might work.

Ben

**Reply by drfinn15** [reply]
Hi Ben,
sorry for my late response. Can you give me any advice on how to do that? The only thing I found out is to do some subnetting on my own, but I am really no professional on that topic. I would appreciate any tip you might have.
Thanks!

**Comment by mKeRix** [comment]
room-assistant also offers a pure [MQTT integration](https://www.room-assistant.io/integrations/mqtt.html) that will push entity updates including all their values (so distances etc.) to a broker. You could hook into this from e.g. Node-RED to do any calculations you want on top of that. If you prefer a pulling setup you can also take a look at the [REST API](https://www.room-assistant.io/guide/api.html), all instances will have all distances listed for their entities there as well.
